src/evaluation_system/api/parameters.py

```diff
@@ -809,10 +811,105 @@ class MyPlugin(plugin.PluginAbstract):
             )
 
     """
+    base_type = datetime
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def parse(self, value: str) -> Union[datetime, None]:
+        """
+        Parse a string value into a datetime object.
+
+        Parameters
+        ----------
+        value : str
+            The string value representing a date.
+
+        Returns
+        -------
+        Union[datetime, None]
+            The parsed datetime object if the parsing is successful,
+            otherwise returns None.
+
+        Raises
+        ------
+        ValueError
+            If the value cannot be converted to a datetime object.
+
+        """
+
+        try:
+            if value:
+                return parse(str(value), default=datetime(1978, 1, 1, 0, 0), fuzzy=False)
+            else:
+                return None
+        except Exception as ex:
+            raise ValueError(f"'{value}' (type {type(value).__name__}) can't be converted to a datetime object") from ex
+
+class DateRange(String):
+    """A tuple containing a date range.
+
+    Parameters
+    ----------
+    kwargs:
+        Additional :class:`ParameterType` parameters.
+
+    Example
+    -------
+
+    .. code-block:: python
+
+        from evaluation_system.api import plugin, parameters
+        class MyPlugin(plugin.PluginAbstract):
+            __parameters__ = parameters.ParameterDictionary(
+                parameters.DateRange(
+                            name="time-range",
+                            default="1950-01-10, 1961",
+                            help="Select a time range separated by comma, YYYY to YYYY-MM-DD HH:mm",
+                            ),
+            )
+
+    """
+
+    base_type = tuple
+
+    def __init__(self,**kwargs):
+        super().__init__(**kwargs)
+        self.dateparse = Date()        
```

Comment:
It's much better if you inherited from Date.

Reply:
thanks